### DISCUSSION TITLE
Error "Workspace not found" on yarn install and yarn add

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2609] [Q&A] [answered]
Hi all! I'm having an issue with yarn add/install with workspaces. I would like to simply add an npm hosted package to one of my workspaces. However, no matter what I try, I always get an error stating my other workspaces listed as dependencies cannot be found. 

My example setup looks like the following. Essentially two packages inside a packages folder and each of the test packages are workspaces. 

.
└── my-app/
    ├── packages/
    │   ├── package-one/
    │   │   └── package.json
    │   └── package-two/
    │       └── package.json
    └── package.json

In my root package.json I have: 

```
 "workspaces": [
    "packages/*"
  ]
```

After running yarn install at the root level and getting everything linked up/installed, I would expect I could `cd` into a package and simply run something like `yarn add lodash`. However, every time I do this I get the error above. It seems like the only thing that works is to add the package name and version manually and then run yarn install at the root level. 

Can someone explain how I could just run something like `yarn add lodash` to install a package in a workspace? 

Additionally, I tried both the `yarn workspace` command and `yarn workspace focus` command to do this. I got the same error in both cases.

**Comment by merceyz** [comment, marked as answer]
Please provide a reproduction, you most likely have a problem in your setup and this isn't enough information to figure out what

**Reply by wdfinch** [reply]
Hi @merceyz. Thanks for the reply. I've attached a simple project I just created to show the issue. I've also attached a screen recording as well.

https://user-images.githubusercontent.com/36898266/111677162-306df700-87dc-11eb-8fe5-25d2277d7cb3.mp4
[yarn_add_issue.zip](https://github.com/yarnpkg/berry/files/6166434/yarn_add_issue.zip)

Really appreciate any guidance you can offer. I'm totally new to yarn 2 and migrating from yarn 1.

**Reply by merceyz** [reply]
As expected you have multiple lockfiles, this means that depending on where you run the command it will either find the other workspaces or not. The lockfile is considered the root of the project so if you reference anything outside of that it can't find it. Remove the nested lockfiles and it should work. In the future we'll throw an error for this so that is made clear

**Reply by wdfinch** [reply]
Gotcha. That makes sense. As a new user, my intuition is to yarn init in each new workspace directory that I add. I noticed that yarn init generates an empty yarn.lock file. I believe in the past this command would only generate the package.json file. 

I guess a few things that could help here aside from a better error message are:
1. Don't generate the empty lock file - instead, let yarn install/add do this
2. Make a note about that on the docs for the cli entries for yarn add and yarn install
